ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -212,9 +293,18 @@ gblBindingTypeSigToEdit GlobalBindingTypeSig{..} mmp
     -- If `mmp` is `Nothing`, return the original range, it used by lenses from diagnostic,
     -- otherwise we apply `toCurrentRange`, and the guard should fail if `toCurrentRange` failed.
     , Just range <- maybe (Just beforeLine) (flip toCurrentRange beforeLine) mmp
-    = Just $ TextEdit range $ T.pack gbRendered <> "\n"
+    -- We need to flatten the signature, as otherwise long signatures are
+    -- rendered on multiple lines with invalid formatting.
```

Comment:
What specifically goes wrong? Should we instead fix the formatting? Do multi-line lenses even work?

Reply:
The lenses display was never multiline, just the textedit was. Not sure why, possibly because the lens display stuff already squishes it into a line. You should be able to see the problem if you remove any long signature from a function, and use type lenses to get it back. The lines following the first line aren't indented, which is illegal syntax.

Reply:
Can confirm, that behaviour has bitten me many times before.

Reply:
Can we at least add a test for this?